beacon-chain/sync/validate_data_column.go

```diff
@@ -141,17 +143,22 @@ func (s *Service) validateDataColumn(ctx context.Context, pid peer.ID, msg *pubs
 	}
 
 	// Get the time at slot start.
-	startTime, err := slots.ToTime(uint64(s.cfg.chain.GenesisTime().Unix()), ds.SignedBlockHeader.Header.Slot)
+	startTime, err := slots.ToTime(uint64(s.cfg.chain.GenesisTime().Unix()), roDataColumn.SignedBlockHeader.Header.Slot)
 	if err != nil {
 		return pubsub.ValidationIgnore, err
 	}
 
-	verifiedRODataColumn, err := verifier.VerifiedRODataColumn()
+	verifiedRODataColumns, err := verifier.VerifiedRODataColumns()
 	if err != nil {
 		return pubsub.ValidationReject, err
 	}
 
-	msg.ValidatorData = verifiedRODataColumn
+	if len(verifiedRODataColumns) != 1 {
```

Comment:
This should be an error logged since its an impossible condition

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/14559/commits/843b43d309a9205a08dd55c43603a866ad81b133.